src/backend/storage/smgr/md.c

```diff
@@ -489,6 +489,8 @@ mdopenfork(SMgrRelation reln, ForkNumber forknum, int behavior)
 	path = relpath(reln->smgr_rnode, forknum);
 
 	fd = PathNameOpenFile(path, O_RDWR | PG_BINARY);
+	if (fd < 0 && (behavior & EXTENSION_CREATE))
```

Comment:
Why this is needed? Needs a comment.

Reply:
It is problem in different place (error-driven programming:)
The trick with EXTENSION_RETURN_NULL allows to eliminate error when we try to read unlogged relation
But once we want to insert something to this relation we need file it to be created.
I will add comment.